securemail/securemail.php

```diff
@@ -1,92 +1,175 @@
-    $s .= '</div><div class="clear"></div>';
-
-    $s .= '<div class="settings-submit-wrapper" ><input type="submit" id="securemail-submit" name="securemail-submit" class="settings-submit" value="' . t('Save Settings') . '" /></div>';
-    $s .= '</div>';
-
-    return;
-}
-function securemail_settings_post(&$a, &$b){
-
-    if(! local_user())
-        return;
-
-    if($_POST['securemail-submit']) {
-		set_pconfig(local_user(),'securemail','pkey',trim($_POST['securemail-pkey']));
-		$enable = ((x($_POST,'securemail-enable')) ? 1 : 0);
-		set_pconfig(local_user(),'securemail','enable', $enable);
-		info( t('Secure Mail Settings saved.') . EOL);
-    }
-}
-
-function securemail_emailer_send_prepare(&$a, &$b) {
-    if (!x($b,'uid')) return;
-	$uid = $b['uid'];
-
-    $enable_checked = get_pconfig($uid,'securemail','enable');
-    if (!$enable_checked) return;
-
-    $public_key_ascii = get_pconfig($uid,'securemail','pkey');
-
-    $gpg = new GPG();
-
-    # create an instance of a GPG public key object based on ASCII key
-    $pub_key = new GPG_Public_Key($public_key_ascii);
-
-    # using the key, encrypt your plain text using the public key
-    $txt_encrypted = $gpg->encrypt($pub_key,$b['textVersion']);
-    #$html_encrypted = $gpg->encrypt($pub_key,$b['htmlVersion']);
-
-    $b['textVersion'] = $txt_encrypted;
-    $b['htmlVersion'] = null;
-}
+<?php
+/**
+ * Name: Secure Mail
+ * Description: Send notification mail encrypted with user-defined public GPG key
+ * Version: 2.0
+ * Author: Fabio Comuni <http://kirgroup.com/profile/fabrixxm>
+ */
+
+require_once "include/Emailer.php";
+
+/* because the fraking openpgp-php is in composer, require libs in composer
+ * and then don't use autoloader to load classes... */
+$path = __DIR__."/vendor/phpseclib/phpseclib/phpseclib/";
+set_include_path(get_include_path() . PATH_SEPARATOR . $path);
+/* so, we don't use the autoloader and include what we need */
+$path = __DIR__."/vendor/singpolyma/openpgp-php/lib";
+set_include_path(get_include_path() . PATH_SEPARATOR . $path);
+
+require_once "openpgp.php";
+require_once "openpgp_crypt_symmetric.php";
+
+
+function securemail_install() {
+    register_hook('plugin_settings', 'addon/securemail/securemail.php', 'securemail_settings');
+    register_hook('plugin_settings_post', 'addon/securemail/securemail.php', 'securemail_settings_post');
+
+    register_hook('emailer_send_prepare',  'addon/securemail/securemail.php', 'securemail_emailer_send_prepare');
+
+    logger("installed securemail");
+}
+
+function securemail_uninstall() {
+    unregister_hook('plugin_settings', 'addon/securemail/securemail.php', 'securemail_settings');
+    unregister_hook('plugin_settings_post', 'addon/securemail/securemail.php', 'securemail_settings_post');
+
+    unregister_hook('emailer_send_prepare',  'addon/securemail/securemail.php', 'securemail_emailer_send_prepare');
+
+    logger("removed securemail");
+}
+
+
+function securemail_settings(&$a,&$s){
```

Comment:
And the `$a` is the `App` class instance. Could be useful to link to it's docs, too...